ssl/record/rec_layer_s3.c

```diff
@@ -356,7 +356,7 @@ int ssl3_write_bytes(SSL *s, int type, const void *buf_, size_t len,
      * promptly send beyond the end of the users buffer ... so we trap and
      * report the error in a way the user will notice
      */
-    if (len < s->rlayer.wnum) {
+    if ((len < s->rlayer.wnum) || ((tot != 0) && (len < (tot + s->rlayer.wpend_tot)))) {
```

Comment:
Why do you check for `tot != 0`? Also I would expect a check for `wb->left != 0` because I *think* it is possible for `wpend_tot` to be stale if `wb->left == 0`. Finally it looks strange to mix use of `tot` and `wnum` on this line because we set `tot = s->rlayer.wnum` just above here.